backend/core/tests/test_helpers.py

```
@@ -143,3 +157,179 @@ def test_get_rating_options_abbr_perm_to_view_matrix():
+    # They are equal at the 'minor' level (i.e. [1, 2] == [1, 2]).
+    compare_schema_versions(schema_ver_a, version_a, version_b, schema_ver_b=1)
+
+
+# Test case 7: Edge case – invalid semantic version string.
+def test_invalid_semver_format():
+    schema_ver_a = None
+    version_a = (
+        "invalid_version"  # This should trigger a formatting error in compare_versions.
+    )
+    version_b = "1.2.3"
+    with pytest.raises(Exception):
```

Ruff (0.8.2)

`pytest.raises(Exception)` should be considered evil

(B017)